include/mega/sync.h

```diff
@@ -26,31 +26,44 @@
 #include "megaclient.h"
 
 namespace mega {
+
+// Returns true for a path that can be synced (.debris is not one of those).
+bool isPathSyncable(const string& localpath, const string& localdebris, const string& localseparator);
+
+// Invalidates the fs IDs of all local nodes below `l` and removes them from `fsidnodes`.
+void invalidateFilesystemIds(handlelocalnode_map& fsidnodes, LocalNode& l, size_t& count);
+
+// Searching from the back, this function compares path1 and path2 character by character and
+// returns the number of consecutive character matches (excluding separators) but only including whole node names.
+// It's assumed that the paths are normalized (e.g. not contain ..) and separated with the given `localseparator`.
+int computeReversePathMatchScore(const string& path1, const string& path2, const string& localseparator);
+
+// Recursively iterates through the filesystem tree starting at the sync root and assigns
+// fs IDs to those local nodes that match the fingerprint retrieved from disk.
+bool assignFilesystemIds(Sync& sync, MegaApp& app, FileSystemAccess& fsaccess, handlelocalnode_map& fsidnodes,
+                         const string& localdebris, const string& localseparator, bool followsymlinks);
+
 class MEGA_API Sync
 {
 public:
-    void *appData;
+    void* appData = nullptr;
 
-    MegaClient* client;
+    MegaClient* client = nullptr;
```

Comment:
client can never actually be null?

Reply:
Has to be null initialized. Similar arguments as for `Node`.